File: 2556-disconnect-path-in-a-binary-matrix-by-at-most-one-flip/2556-disconnect-path-in-a-binary-matrix-by-at-most-one-flip.ts
function isPossibleToCutPath(grid: number[][]): boolean {
    let n : number = grid.length;
    let m : number = grid[0].length;
    
    let arr : number[] = new Array(m+n-1).fill(0);
    
    for(let i = 0; i<n; i++){
        for(let j = 0; j<m; j++){
            if(grid[i][j] == 1)
                arr[i+j]++;
        }
    }
    
    for(let i = 1; i < arr.length-1; i++){
        if(arr[i] <= 1) return true;
    }
    return false;
};

File: 2556-disconnect-path-in-a-binary-matrix-by-at-most-one-flip/2556-disconnect-path-in-a-binary-matrix-by-at-most-one-flip.cpp
class Solution {
public:
    bool isPossibleToCutPath(vector<vector<int>>& grid) {
        
        int n = grid.size();
        int m = grid[0].size();
        
        vector<int>v(m+n-1,0);
        
        for(int i = 0; i<n; i++){
            for(int j = 0; j<m; j++){
                if(grid[i][j])
                    v[i+j]++;
            }
        }
        
        for(int i = 1; i < v.size()-1; i++){
            if(v[i] <= 1) return true;
        }
        
        return false;
    }
};